Objects/longobject.c

```diff
@@ -6212,6 +6212,25 @@ int___round___impl(PyObject *self, PyObject *o_ndigits)
         return long_long(self);
     }
 
+    int64_t n, e;
+
+    if (PyLong_AsInt64(ndigits, &n) < 0) {
+        Py_DECREF(ndigits);
+        return NULL;
+    }
+    n = -n;
+
+    /* A quick exit, if the result is zero.
+     * We start from criteria abs(self) < 10**(n - 1).  Then, from
+     * definition of the _PyLong_Frexp() we have
+     * abs(self) < 10*10**(log10(2)*e) < 10**(e/3 + 1). */
+    (void)_PyLong_Frexp((PyLongObject *)self, &e);
```

Comment:
```suggestion
    (void)_PyLong_Frexp((PyLongObject *)self, &e);
    assert(e >= 0);
    assert(!PyErr_Occurred());
```

in that PR most usages are still checked with `assert`s

Reply:
I think this should be rather documented in the `_PyLong_Frexp()` description.  It already says that `e>=0`, but https://github.com/python/cpython/commit/d08c7888229e78533648191dfe42e2d2d3ecea25 should be amended to remove OverflowError case.

Though, lets include asserts for consistency with other use-cases.